<launch>

    <!-- defines name of the namespace of the drone -->
    <arg name="UAV_NAME" default="$(env UAV_NAME)"/>

    <arg name="simulation" default="true"/>

    <!-- will it run using GNU debugger? -->
    <arg name="DEBUG" default="false"/>
    <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
    <arg if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

    <!-- will it run as standalone nodelet or using a nodelet manager? -->
    <arg name="standalone" default="true"/>
    <arg name="manager" default="$(arg UAV_NAME)_icp_manager"/>
    <arg name="n_threads" default="1"/>
    <arg unless="$(arg standalone)" name="nodelet" value="load"/>
    <arg if="$(arg standalone)" name="nodelet" value="standalone"/>
    <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
    <arg if="$(arg standalone)" name="nodelet_manager" value=""/>

    <!-- Namespace -->
    <group ns="$(arg UAV_NAME)">

        <!-- WaypointFlier nodelet -->
        <node pkg="nodelet" type="nodelet" name="icp_node" args="$(arg nodelet) icp_node/IcpNode $(arg nodelet_manager)"
              launch-prefix="$(arg launch_prefix_debug)" output="screen">

            <!-- Parameters loaded from launch file -->
            <param name="uav_name" value="$(arg UAV_NAME)"/>

            <!-- ROS parameters config file -->
            <rosparam file="$(find icp_node)/config/localization.yaml"/>

            <!-- Subscribers -->
            <remap from="~points_in" to="os_cloud_nodelet/points"/>
            <remap from="~ground_truth" to="ground_truth"/>

            <!-- Map name -->
            <param name="map_frame_id" value="$(arg uav_name)/map_origin"/>
        </node>

    </group>

</launch>
